### DISCUSSION TITLE
Prevent math.import(...) from overriding units

### DISCUSSION THREAD

**Discussion by bornova** [josdejong/mathjs#3222] [Ideas]
Would it be possible to extend the `override: false` option in `math.import(...)` to units as well?

For example, if a function named `N` is imported using `math.import({ N: function () {...} })`, it overrides the `newton` unit `N`.  Would be great to have an option to prevent this.

**Comment by josdejong** [comment]
That makes a lot of sense! 

I've created a PR here: https://github.com/josdejong/mathjs/pull/3225 it's a breaking change though so we'll have to wait until the first next major release to merge and publish it.

As a workaround, you could create your own wrapper function around `math.import` and check whether an import is a unit using `math.Unit.isValuelessUnit(...)` and if so, throw an exception.

**Reply by josdejong** [reply]
The fix is published now in `v14.0.0`.

**Comment by bornova** [comment]
Thanks @josdejong!  I worked around it by importing as factory function with a custom name instead.  So instead of directly importing function `N`, I am importing as `myfunc.N`. This way I got to keep both `N` unit and the 'N' function imported from an external library.  Looking forward to the next major release.

**Reply by josdejong** [reply]
Ah, that is indeed a neat way to prevent this issue, thanks for sharing!
